python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1303 @@
+    def getTargetList(parsedCmd, **kwargs):
+        """Parse the visit list and pass through explicitly."""
+        visitPairs = [(int(visit.split(",")[0]), int(visit.split(",")[1])) for visit in parsedCmd.visitPairs]
+        return pipeBase.TaskRunner.getTargetList(parsedCmd, visitPairs=visitPairs, **kwargs)
+
+
+class BrighterFatterKernelTaskDataIdContainer(pipeBase.DataIdContainer):
+    """A DataIdContainer for the MakeBrighterFatterKernelTask."""
+
+    def makeDataRefList(self, namespace):
+        """Compute refList based on idList.
+
+        This method must be defined as the dataset does not exist before this task is run.
+
+        Parameters
+        ----------
+        namespace
+            Results of parsing command-line (with ``butler`` and ``log`` elements).
+
+        Notes
+        -----
+        Not called if ``add_id_argument`` called with ``doMakeDataRefList=False``.
+        Note that this is almost a copy-and-paste of the vanilla implementation, but without checking
+        if the datasets already exist as this task exists to make them.
+        """
+        if self.datasetType is None:
+            raise RuntimeError("Must call setDatasetType first")
+        butler = namespace.butler
+        for dataId in self.idList:
+            refList = list(butler.subset(datasetType=self.datasetType, level=self.level, dataId=dataId))
+            # exclude nonexistent data
+            # this is a recursive test, e.g. for the sake of "raw" data
+            if not refList:
+                namespace.log.warn("No data found for dataId=%s", dataId)
+                continue
+            self.refList += refList
+
+
+class MakeBrighterFatterKernelTask(pipeBase.CmdLineTask):
+    """Brighter-fatter effect correction-kernel calculation task.
+
+    A command line task for calculating the brighter-fatter correction
+    kernel from pairs of flat-field images (with the same exposure length).
+
+    The following operations are performed:
+
+    - The configurable isr task is called, which unpersists and assembles the
+      raw images, and performs the selected instrument signature removal tasks.
+      The appropriacy of the selected isr components is checked.
+
+    - The gain of the each amplifier in the detector is calculated using
+      the PTC method and used to correct the images so that all calculations
+      are done in units of electrons, and so that the level across amplifier
+      boundaries is continuous. Outliers in the PTC are iteratively rejected
+      before fitting, with the nSigma rejection level set by
+      config.nSigmaClipRegression. Individual pixels are masked from
+      the image based on config.nSigmaClipGainCalc
+
+    - Each image is then cross-correlated with the one it's paired with
+      (with the pairing defined by the --visitPairs command line argument),
+      which is done either the whole-image to whole-image,
+      or amplifier-by-amplifier, depending on the config.level.
```

Comment:
I suggest removing "the" before "config.level", but overall this is excellent -- very clear and helpful